internal/cloud/keyvault.go

```diff
@@ -55,134 +53,103 @@ func getAzureClientAuthMethod() clientAuthMethod {
-func NewKey(client *keyvault.BaseClient, vaultName, keyName, keyVersion string) (*Key, error) {
-	dnssuffix := os.Getenv("AZURE_KEYVAULT_DNSSUFFIX")
-	if dnssuffix == "" {
-		var env azure.Environment
-		if envName := os.Getenv("AZURE_ENVIRONMENT"); envName == "" {
-			env = azure.PublicCloud
-		} else {
-			var err error
-			env, err = azure.EnvironmentFromName(envName)
-			if err != nil {
-				return nil, err
-			}
-		}
-		dnssuffix = env.KeyVaultDNSSuffix
+	certClient, err := azcertificates.NewClient(vaultURL, credential, nil)
+	if err != nil {
+		return nil, err
 	}
-	return &Key{
-		Client:       client,
-		vaultBaseURL: "https://" + vaultName + "." + dnssuffix,
+	secretClient, err := azsecrets.NewClient(vaultURL, credential, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KeyVault{
+		keyClient:    keyClient,
+		certClient:   certClient,
+		secretClient: secretClient,
 		name:         keyName,
-		version:      keyVersion,
+		version:      version,
 	}, nil
 }
 
 // Sign signs the message digest with the algorithm provided.
-func (k *Key) Sign(ctx context.Context, algorithm keyvault.JSONWebKeySignatureAlgorithm, digest []byte) ([]byte, error) {
-	// Prepare the message
-	value := base64.RawURLEncoding.EncodeToString(digest)
-
+func (k *KeyVault) Sign(ctx context.Context, algorithm azkeys.JSONWebKeySignatureAlgorithm, digest []byte) ([]byte, error) {
 	// Sign the message
-	res, err := k.Client.Sign(
+	res, err := k.keyClient.Sign(
 		ctx,
-		k.vaultBaseURL,
 		k.name,
 		k.version,
-		keyvault.KeySignParameters{
-			Algorithm: algorithm,
-			Value:     &value,
+		azkeys.SignParameters{
+			Algorithm: &algorithm,
+			Value:     digest,
 		},
+		nil,
 	)
 	if err != nil {
 		return nil, err
 	}
 
 	// Verify the result
-	if res.Kid == nil {
+	if res.KID == nil {
 		return nil, errors.New("azure: nil kid")
 	}
 	if res.Result == nil {
 		return nil, errors.New("azure: invalid server response")
 	}
-	return base64.RawURLEncoding.DecodeString(*res.Result)
+	return res.Result, nil
 }
 
 // CertificateChain returns the X.509 certificate chain associated with the key.
-func (k *Key) CertificateChain(ctx context.Context) ([]*x509.Certificate, error) {
-	// Need a certificate chain to pass the notation validation
-	// GetCertificate only returns the leaf certificate
-	// So we need a GetSecret to get the full certificate chain
-	res, err := k.Client.GetSecret(
-		ctx,
-		k.vaultBaseURL,
-		k.name,
-		k.version,
-	)
+func (k *KeyVault) CertificateChain(ctx context.Context) ([]*x509.Certificate, error) {
+	secret, err := k.secretClient.GetSecret(ctx, k.name, k.version, nil)
 	if err != nil {
 		return nil, err
 	}
-	if res.Value == nil || res.ContentType == nil {
+	if secret.Value == nil || secret.ContentType == nil {
 		return nil, errors.New("azure: invalid server response")
 	}
-	return crypto.ParseCertificates([]byte(*res.Value), *res.ContentType)
+	return crypto.ParseCertificates([]byte(*secret.Value), *secret.ContentType)
```

Comment:
What's the example content in `secret.Value`? Is it the raw secret or a base64-encoded string of the raw secret?

Reply:
It is the  raw secret.

Reply:
We need to document this behavior somewhere.

Reply:
Added comments.